ts/src/digifinex.ts

```diff
@@ -1553,28 +1554,169 @@ export default class digifinex extends Exchange {
          * @param {string} symbol unified symbol of the market to create an order in
          * @param {string} type 'market' or 'limit'
          * @param {string} side 'buy' or 'sell'
-         * @param {float} amount how much of currency you want to trade in units of base currency
+         * @param {float} amount how much you want to trade in units of the base currency, spot market orders use the quote currency, swap requires the number of contracts
          * @param {float} [price] the price at which the order is to be fullfilled, in units of the quote currency, ignored in market orders
          * @param {object} [params] extra parameters specific to the digifinex api endpoint
          * @param {string} [params.timeInForce] "GTC", "IOC", "FOK", or "PO"
          * @param {bool} [params.postOnly] true or false
          * @param {bool} [params.reduceOnly] true or false
+         * @param {string} [params.marginMode] 'cross' or 'isolated', for spot margin trading
          * @returns {object} an [order structure]{@link https://github.com/ccxt/ccxt/wiki/Manual#order-structure}
          */
         await this.loadMarkets ();
         const market = this.market (symbol);
-        symbol = market['symbol'];
+        const marginResult = this.handleMarginModeAndParams ('createOrder', params);
```

Comment:
weird I can place this order through the Web UI but not with the API 
```
n digifinex createOrder "LTC/USDT" market buy 0.15 --report
Debugger attached.
2023-11-03T15:21:15.131Z
Node.js: v18.18.0
CCXT v4.1.36
digifinex.createOrder (LTC/USDT, market, buy, 0.15)
InvalidOrder Cash Amount is less than the minimum requirement
---------------------------------------------------
[InvalidOrder] Cash Amount is less than the minimum requirement

    at handleErrors               Users/cjg/Git/ccxt10/ccxt/js/src/digifinex.js:4202      
    at                            Users/cjg/Git/ccxt10/ccxt/js/src/base/Exchange.js:777   
    at processTicksAndRejections  node:internal/process/task_queues:95                    
    at fetch2                     Users/cjg/Git/ccxt10/ccxt/js/src/base/Exchange.js:2856  
    at request                    Users/cjg/Git/ccxt10/ccxt/js/src/base/Exchange.js:2859  
    at createOrder                Users/cjg/Git/ccxt10/ccxt/js/src/digifinex.js:1575      
    at async run                  Users/cjg/Git/ccxt10/ccxt/examples/js/cli.js:315        

Cash Amount is less than the minimum requirement
```

Reply:
also can't place swap orders
```
n digifinex createOrder "LTC/USDT:USDT" limit buy 0.1 55 --report
Debugger attached.
2023-11-03T15:23:41.624Z
Node.js: v18.18.0
CCXT v4.1.36
digifinex.createOrder (LTC/USDT:USDT, limit, buy, 0.1, 55)
BadRequest Invalid Parameter
---------------------------------------------------
[BadRequest] Invalid Parameter

    at handleErrors               Users/cjg/Git/ccxt10/ccxt/js/src/digifinex.js:4202      
    at                            Users/cjg/Git/ccxt10/ccxt/js/src/base/Exchange.js:777   
    at processTicksAndRejections  node:internal/process/task_queues:95                    
    at fetch2                     Users/cjg/Git/ccxt10/ccxt/js/src/base/Exchange.js:2856  
    at request                    Users/cjg/Git/ccxt10/ccxt/js/src/base/Exchange.js:2859  
    at createOrder                Users/cjg/Git/ccxt10/ccxt/js/src/digifinex.js:1568      
    at async run                  Users/cjg/Git/ccxt10/ccxt/examples/js/cli.js:315        
```

Reply:
ok we're missing the createMarketBuyRequiresPrice logic
![image](https://github.com/ccxt/ccxt/assets/43336371/187dabb8-1516-4d53-9062-e3d106bd6e35)